### PR TITLE
Switch examples to Webpack

### PR DESCRIPTION
**Pull request by tdawes**
And add source-maps and HMR.

Built on top of #79

### CODE REVIEW

File: examples/chat-app/package.json [hunk 1]
```diff
@@ -29,10 +29,21 @@
     "@types/react": "^16.8.25",
     "@types/react-dom": "^16.8.5",
     "@types/testing-library__react": "^9.1.1",
+    "@types/webpack-env": "^1.14.0",
+    "babel-loader": "^8.0.6",
+    "css-loader": "^3.2.0",
+    "fork-ts-checker-webpack-plugin": "^1.5.0",
+    "html-webpack-plugin": "^3.2.0",
     "jest": "^24.9.0",
     "parcel-bundler": "^1.12.3",
```
Comment (coffee-cup) [inline examples/chat-app/package.json]: Can remove this

File: examples/chat-app/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    path: path.resolve(__dirname, "dist"),
+    publicPath: "/",
+  },
+
+  resolve: {
+    extensions: [".ts", ".tsx", ".js", ".json"],
+  },
+
+  devtool: "source-map",
+
```
Comment (coffee-cup) [inline examples/chat-app/webpack.config.js]: 🔫

File: examples/chat-app/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        test: /\.(ts|js)x?$/,
+        use: ["babel-loader", "source-map-loader"],
+        exclude: /node_modules/,
+      },
+      {
+        test: /\.scss$/,
+        loader: ["style-loader", "css-loader", "sass-loader"],
+      },
+      {
+        test: /\.css$/i,
```
Comment (coffee-cup) [inline examples/chat-app/webpack.config.js]: I think the css and scss loader can be combined

File: examples/clock/package.json [hunk 4]
```diff
@@ -23,12 +23,24 @@
...
+    "@types/webpack-env": "^1.14.0",
+    "babel-loader": "^8.0.6",
     "babel-preset-env": "^1.7.0",
     "babel-preset-react": "^6.24.1",
+    "css-loader": "^3.2.0",
+    "fork-ts-checker-webpack-plugin": "^1.5.0",
+    "html-webpack-plugin": "^3.2.0",
     "jest": "^24.9.0",
+    "node-sass": "^4.12.0",
     "parcel-bundler": "^1.12.3",
```
Comment (coffee-cup) [inline examples/clock/package.json]: Can remove (as well with all other examples)

File: examples/clock/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  entry: APP_PATH,
+
+  output: {
+    filename: "bundle.js",
+    path: path.resolve(__dirname, "dist"),
+    publicPath: "/",
+  },
+
+  resolve: {
+    extensions: [".ts", ".tsx", ".js", ".json"],
```
Comment (coffee-cup) [inline examples/clock/webpack.config.js]: `.jsx` too maybe?

File: examples/todo-app/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      },
+      {
+        test: /\.css$/i,
+        use: ["style-loader", "css-loader"],
+      },
+      {
+        test: /\.(png|jpe?g|svg|ico)$/,
+        loader: "file-loader",
+        options: {
+          publicPath: "/",
```
Comment (coffee-cup) [inline examples/todo-app/webpack.config.js]: Do we need both this `publicPath` and the `publicPath` above?

Comment (tdawes) [reply examples/todo-app/webpack.config.js]: I think so - will try and remove and see

**Pull request merged**